src-input/duk_numconv.c

```diff
@@ -93,12 +93,12 @@ DUK_LOCAL void duk__bi_print(const char *name, duk__bigint *x) {
 	duk_small_int_t i;
 
 	/* No NUL term checks in this debug code. */
-	p += DUK_SPRINTF(p, "%p n=%ld", (void *) x, (long) x->n);
+	p += DUK_SNPRINTF(p, sizeof(buf), "%p n=%ld", (void *) x, (long) x->n);
```

Comment:
Using the return value of snprintf() is a bit dangerous assuming one actually prepares for truncation: the return value indicates what would have been written if truncation wasn't needed -- not the number of bytes actually written (i.e. the truncated size). So this only really works if the buffer is not overrun :-/.

This snprintf() behavior actually makes it quite awkward to accumulate data into a buffer safely because one would need to do so without using the snprintf() return value, or by comparing the return value after every snprintf() to the buffer size given as snprintf() argument (retval >= size means truncation).

Reply:
This code is used for debugging, so I refactored it to make it safer without assuming snprintf does the right things. :)